src/formatDate.js

```diff
@@ -9,6 +9,81 @@
  */
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+  let newDate = '';
+  let fromCharacter = 0;
+  let month;
+  let day;
+  let year;
+  let yearFormat;
+
+  fromFormat.forEach((element) => {
+    const toCharacter = fromCharacter + element.length;
+    const separator = fromFormat[fromFormat.length - 1];
+    const currentCharacter = fromCharacter + element.length + separator.length;
+
+    if (element === 'MM') {
+      month = date.slice(fromCharacter, toCharacter);
+      fromCharacter = currentCharacter;
+    }
+
+    if (element === 'DD') {
+      day = date.slice(fromCharacter, toCharacter);
+      fromCharacter = currentCharacter;
+    }
+
+    if (element === 'YYYY') {
+      year = date.slice(fromCharacter, toCharacter);
+      fromCharacter = currentCharacter;
+      yearFormat = element;
+    }
+
+    if (element === 'YY') {
+      year = date.slice(fromCharacter, toCharacter);
+      fromCharacter = currentCharacter;
+      yearFormat = element;
+    }
```

Comment:
The logic for extracting date components is based on a fixed separator length, which assumes that the separator is always one character long. This could lead to incorrect slicing if the separator is more than one character. Instead, consider using the last item in the `fromFormat` array directly as the separator, and adjust the slicing logic accordingly.